lang_GENERIC/analyze/Naming_AST.ml

```diff
@@ -254,6 +254,30 @@ let lookup_nonlocal_scope id scopes =
       None
 (*e: function [[Naming_AST.lookup_nonlocal_scope]] *)
 
+let make_type type_string =
+  Some(TyName(((type_string, Parse_info.fake_info "()"), {name_qualifier=None; name_typeargs=None})))
+
+let get_resolved_type (vinit, vtype) =
+  match vtype with
+    | Some(_) -> vtype
+    | None -> (
+      match vinit with
+         | Some(L (Bool _)) -> make_type "bool"
```

Comment:
Maybe can add a comment saying an alternative would be to define a TyInt, TyBool in the generic AST
so this code would be more portable across languages. We would not need to do a 
switch on the language.